### PR TITLE
🌐 Add French translation for `docs/fr/docs/tutorial/tests.md`

### PR DESCRIPTION
**Pull request by Mgcsuper**
A possible translation in frensh of the doc file testing.md

### CODE REVIEW

**Comment by tiangolo** [comment]
📝 Docs preview for commit 8d6b290d0943c286f72555bc2815577c3ff37edf at: https://a0d7986f.fastapitiangolo.pages.dev

**Comment by tiangolo** [comment]
📝 Docs preview for commit a171dbd3f50465984c7b221a012f8f3d359f8ac4 at: https://f9b182bf.fastapitiangolo.pages.dev

**Comment by Mgcsuper** [comment]
I don't know how to do two different pull requests

**Comment by tiangolo** [comment]
📝 Docs preview for commit d49ee02f440787d064c6fdddf3468512c92ffcd0 at: https://cde026da.fastapitiangolo.pages.dev

**Comment by tiangolo** [comment]
📝 Docs preview for commit ab36385ded0135479d421010c96a2a06ab217d44 at: https://962b54f3.fastapitiangolo.pages.dev

**Comment by tiangolo** [comment]
📝 Docs preview for commit 04ec885accdf00f6f820dae9a1d45c2ec8b9eda4 at: https://1f7cc2c4.fastapitiangolo.pages.dev

**Comment by tiangolo** [comment]
📝 Docs preview for commit dab148cba8f4b093174d66c2b0057943810a6d4a at: https://21a82c31.fastapitiangolo.pages.dev

File: docs/fr/docs/tutorial/tests.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+Importez `TestClient`.
+
+Créez un `TestClient` en lui passant votre application **FastAPI**.
+
+Créez des fonctions dont le nom commence par `test_` (c'est une convention standard de `pytest`).
+
+Utilisez l'objet `TestClient` de la même manière que vous le feriez avec `httpx`.
+
+Écrivez des instruction d'assertion `assert` simples avec les expressions Python standard qu'il est necessaire de vérifier (de nouveau, c'est une pratique standard de `pytest`)."
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:23]:
```suggestion
Écrivez des instructions d'assertion `assert` simples avec les expressions Python standard qu'il est necessaire de vérifier (de nouveau, c'est une pratique standard de `pytest`)."
```

File: docs/fr/docs/tutorial/tests.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+```Python hl_lines="2  12  15-18"
+{!../../../docs_src/app_testing/tutorial001.py!}
+```
+
+!!! astuce
+    il est utile de remarquer que les fonction de test sont normal `def`, et non `async def`
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:30]:
```suggestion
    il est important de remarquer que les fonction de tests sont normal `def`, et non `async def`
```

File: docs/fr/docs/tutorial/tests.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+    Et les appel au client sont aussi normaux, pas besoin d'utiliser `await`
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:32]:
```suggestion
    Les appels au client sont aussi normaux, pas besoin d'utiliser `await`
```

File: docs/fr/docs/tutorial/tests.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+
+!!! note "Details Technique"
+    il est aussi possible d'utiliser `from starlette.testclient import TestClient`.
+
+    **FastAPI** fournit la même `starlette.testclient` que `fastapi.testclient` pour une simple question de commodité pour vous, développeur. Mais provient directement de Starlette.
+
+
+!!! astuce
+    Si vous chercher tout de même à utiliser la foncion `async` dans vos test, à l'exeption d’envoyer des requêtes à votre application FastAPI (par exemple, des fonctions de base de données asynchrones), jetez un coup d’œil aux [Tests asynchrones (en)](../advanced/async-tests.md) {.internal-link target=_blank} dans le tutoriel avancé.
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:44]:
```suggestion
        Si vous devez utiliser `async` dans vos tests (par exemple, pour des fonctions de base de données asynchrones), sauf pour envoyer des requêtes à votre application FastAPI, consultez les [Tests asynchrones](../advanced/async-tests.md) {.internal-link target=_blank} dans le tutoriel avancé.
.

```

File: docs/fr/docs/tutorial/tests.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+## tests séparé
+
+Dans une application réelle, vous auriez probablement vos tests dans un fichier différent.
+
+Et votre application **FastAPI** peut également être composée de plusieurs fichiers/modules, etc.
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:50]:
```suggestion
Votre application **FastAPI** peut également être composée de plusieurs fichiers/modules, etc.
```

File: docs/fr/docs/tutorial/tests.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+### fichier app de **FastAPI**
+
+disons que vous ayez une structure de fichier comme décrit dans [application de grande taille (en)](./bigger-applications.md) {.internal-link target=_blank}:
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:54]:
```suggestion
Disons que vous ayez une structure de fichier comme décrit dans [application de grande taille (en)](./bigger-applications.md) {.internal-link target=_blank}:
```

File: docs/fr/docs/tutorial/tests.md [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+!!! astuce
+    Si vous chercher tout de même à utiliser la foncion `async` dans vos test, à l'exeption d’envoyer des requêtes à votre application FastAPI (par exemple, des fonctions de base de données asynchrones), jetez un coup d’œil aux [Tests asynchrones (en)](../advanced/async-tests.md) {.internal-link target=_blank} dans le tutoriel avancé.
+
+## tests séparé
+
+Dans une application réelle, vous auriez probablement vos tests dans un fichier différent.
+
+Et votre application **FastAPI** peut également être composée de plusieurs fichiers/modules, etc.
+
+### fichier app de **FastAPI**
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:52]:
```suggestion
### Fchier app de **FastAPI**
```

File: docs/fr/docs/tutorial/tests.md [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+disons que vous ayez une structure de fichier comme décrit dans [application de grande taille (en)](./bigger-applications.md) {.internal-link target=_blank}:
+
+```
+.
+├── app
+│   ├── __init__.py
+│   └── main.py
+```
+
+dans le fichier `main.py` vous avez votre application **FastAPI** :
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:63]:
```suggestion
Le fichier `main.py` vous permet d'accéder à votre application **FastAPI** :
```

File: docs/fr/docs/tutorial/tests.md [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+
+
+```Python
+{!../../../docs_src/app_testing/main.py!}
+```
+
+### fichier de test
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:70]:
```suggestion
### Fichier de test
```

File: docs/fr/docs/tutorial/tests.md [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+
+```Python
+{!../../../docs_src/app_testing/test_main.py!}
+```
+
+...et obtenir le code pour les tests comme avant.
+
+## Test: exemple supplémentaire
+
+Allons plus loin dans cet exemple et ajoutons plus de détails pour voir comment tester différentes pièces.
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:92]:
```suggestion
Pour aller plus loin dans cet exemple, ajoutons plus de détails pour observer comment tester différentes pièces.
```

File: docs/fr/docs/tutorial/tests.md [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+
+### fichier app étendu de **FastAPI**
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:94]:
```suggestion
### Fichier App étendu de **FastAPI**
```

File: docs/fr/docs/tutorial/tests.md [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+
+```
+.
+├── app
+│   ├── __init__.py
+│   ├── main.py
+│   └── test_main.py
+```
+
+Supposons que maintenant le fichier `main.py` avec votre application **FastAPI** a d’autres **opérations sur chemin d'accès**.
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:106]:
```suggestion
Supposons que maintenant le fichier `main.py` avec votre application **FastAPI** à d’autres **opérations sur chemin d'accès**.
```

File: docs/fr/docs/tutorial/tests.md [hunk 13]
```diff
@@ -0,0 +1,216 @@
...
+
+Il dispose d’une opération `GET` qui pourrait renvoyer une erreur.
+
+Et dispose d’une opération `POST` qui peut renvoyer plusieurs erreurs.
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:110]:
```suggestion
Il  dispose d’une opération `POST` qui peut renvoyer plusieurs erreurs.
```

File: docs/fr/docs/tutorial/tests.md [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+=== "Python 3.8+ non annoté"
+
+!!! astuce
+Favoriser l'utilisation de la version `Annoté` si possible.
+
+```Python
+{!> ../../../docs_src/app_testing/app_b/main.py!}
+```
+
+### fichier de test étendu
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:150]:
```suggestion
### Fichier de test étendu
```

File: docs/fr/docs/tutorial/tests.md [hunk 15]
```diff
@@ -0,0 +1,216 @@
...
+* Si vous avez besoin d’envoyer *Form Data* au lieu de JSON, utilisez le paramètre `data` à la place.
+
+* Pour passer des *headers*, utilisez un `dict` dans le paramètre `headers`.
+
+* Pour les *cookies*, un `dict` dans le paramètre `cookies`.
+
+Pour plus d’informations sur la façon de passer des données au backend (en utilisant `httpx` ou `TestClient`), consultez la <a href="https://www.python-httpx.org" class="external-link" target="_blank">documentation HTTPX</a>.
+
+!!! information
+    à otez que le `TestClient` reçoit des données qui peuvent être converties en JSON, et non en modèles Pydantic.
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:176]:
```suggestion
    à noter que le `TestClient` reçoit des données qui peuvent être converties en JSON, et non en modèles Pydantic.
```

File: docs/fr/docs/tutorial/tests.md [hunk 16]
```diff
@@ -0,0 +1,216 @@
...
+
+```console
+$ pip install pytest
+
+---> 100%
+```
+</div>
+
+
+Cela détectera automatiquement les fichiers et les tests, les exécutera et vous rapportera les résultats.
```
Comment (GDemay) [inline docs/fr/docs/tutorial/tests.md:194]:
```suggestion
Cela détectera automatiquement les fichiers et les tests, les exécutera, puis vous rapportera les résultats.
```

**Review by GDemay — CHANGES_REQUESTED** [review]
I improved the translation and fixed the errors. To be transparent, I think there are still some mistakes in grammar, accents, etc., but it's a good improvement.

**Comment by GDemay** [comment]
Why does the smokeshow / coverage at 99% ? Can we start a new build? @Mgcsuper

**Pull request closed without merging**